tests/PerformanceTests/PerformanceTestService/EventSource/BenchmarkMsalDistributedTokenCacheAdapter.cs

```diff
@@ -25,10 +26,17 @@ public class BenchmarkMsalDistributedTokenCacheAdapter : MsalDistributedTokenCac
         /// </summary>
         /// <param name="cacheKey">Key of the cache to remove.</param>
         /// <returns>A <see cref="Task"/> that completes when key removal has completed.</returns>
-        protected override async Task RemoveKeyAsync(string cacheKey)
+        protected override Task RemoveKeyAsync(string cacheKey)
         {
+            var bytes = base.ReadCacheBytesAsync(cacheKey).GetAwaiter().GetResult();
+
+            if (bytes != null)
+            {
+                MemoryCacheEventSource.Log.DecrementSize(bytes.Length);
+            }
```

Comment:
what is the cpu cost of doing this?

Reply:
As per docs, EventSource is low overhead. AppInsights SDK gets the data every 60 seconds. If we want to collect periodically, there's no way to directly get item count and size from the memory caches, so seems like we would still have to have some sort of a counter running.